src/Providers/PayoneServiceProvider.php

```diff
@@ -337,21 +339,34 @@ protected function registerPaymentExecute(Dispatcher $dispatcher, BasketReposito
                     $paymentCache = pluginApp(PaymentCache::class);
 
                     $order = $orderRepository->findOrderById($event->getOrderId());
-                    $payment = $paymentCache->loadPayment($event->getMop());
-                    if (!($payment instanceof Payment)) {
-                        $message = 'Payment could not be assigned to order.';
-
-                        /** @var Logger $logger */
-                        $logger = pluginApp(Logger::class);
-                        $logger->error($message, $payment);
-                        return;
+                    if($order instanceof Order) {
+                        if($event->getMop() == $paymentHelper->getMopId(PayoneInvoiceSecurePaymentMethod::PAYMENT_CODE)) {
+                            //Block the invoice generation for secure invoice because there will be an external invoice
+                            $orderRepository->updateOrder([
+                                'properties' => [
+                                    [
+                                        'typeId' => OrderPropertyType::EXTERNAL_TAX_SERVICE,
+                                        'value' => 1
```

Comment:
Is this the correct way to do this? What about locking the order?

Reply:
This is a way which is also announced in our forum 😄 
But nevertheless i'll have a call with Aye and maybe Payone to specify the process. I also have some question to the whole workflow. Come back if i had more informations.

Reply:
Ok now it is clear. 
So currently we only want to block the invoice generation because there will be an external invoice.